pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -7501,6 +7513,85 @@ spec:
 	}
 }
 
+func Test_taskWorkspaceByWorkspaceVolumeSource(t *testing.T) {
+	testPr := &v1beta1.PipelineRun{}
+	tests := []struct {
+		name, taskWorkspaceName, pipelineWorkspaceName, prName string
+		wb                                                     v1beta1.WorkspaceBinding
+		expectedBinding                                        v1beta1.WorkspaceBinding
+		disableAffinityAssistant                               bool
+	}{
+		{
+			name:                  "PVC Workspace with Affinity Assistant",
+			prName:                "test-pipeline-run",
+			taskWorkspaceName:     "task-workspace",
+			pipelineWorkspaceName: "pipeline-workspace",
+			wb: v1beta1.WorkspaceBinding{
+				Name:                "foo",
+				VolumeClaimTemplate: &corev1.PersistentVolumeClaim{},
+			},
+			expectedBinding: v1beta1.WorkspaceBinding{
+				Name: "task-workspace",
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "pvc-2c26b46b68-affinity-assistant-e011a5ef79-0",
+				},
+			},
+		},
+		{
+			name:              "PVC Workspace without Affinity Assistant",
+			prName:            "test-pipeline-run",
+			taskWorkspaceName: "task-workspace",
+			wb: v1beta1.WorkspaceBinding{
+				Name:                "foo",
+				VolumeClaimTemplate: &corev1.PersistentVolumeClaim{},
+			},
+			expectedBinding: v1beta1.WorkspaceBinding{
+				Name: "task-workspace",
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "pvc-2c26b46b68",
+				},
+			},
+			disableAffinityAssistant: true,
+		},
+		{
+			name:              "non-PVC Workspace",
+			taskWorkspaceName: "task-workspace",
+			wb: v1beta1.WorkspaceBinding{
+				Name:     "foo",
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+			expectedBinding: v1beta1.WorkspaceBinding{
+				Name:     "task-workspace",
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			c := Reconciler{}
+			ctx := context.Background()
+			if tc.disableAffinityAssistant {
+				featureFlags, err := config.NewFeatureFlagsFromMap(map[string]string{
+					"disable-affinity-assistant": "true",
```

Comment:
nit: would it be simpler to create the configmap in all cases, and have the value of `disable-affinity-assistant` be passed in from the configmap?

Reply:
Hmm I agree that putting the configmap in the tests would help when we want to set multiple feature flags. Since `disable-affinity-assistant` is the only related field, I think a boolean flag in the test case would be easier to read instead?

Reply:
yeah a boolean flag is what I had in mind, I was thinking something like:
```
featureFlags, err := config.NewFeatureFlagsFromMap(map[string]string{
    "disable-affinity-assistant": strconv.FormatBool(tc.disable-affinity-assistant),
}
```